src/util.rs

```diff
@@ -89,6 +90,46 @@ pub(crate) fn uname_release() -> Result<CString> {
     Ok(release)
 }
 
+pub(crate) fn find_lower_bound_by<T, F>(slice: &[T], mut f: F) -> Option<usize>
```

Comment:
lower_bound is misleading, as "lower_bound" implies that you can find element that's not exactly equal to the one you were searching for. So this is really an exact binary search with a guarantee to find leftmost element. `find_leftmost` or `find_equal_leftmost` probably would be a bit less confusing names?

Reply:
Actually, just noticed `find_match_or_lower_bound`, so with that terminology this would be named `find_leftmost_match` or `find_lowest_match`, right?

Reply:
I went with the naming based on https://github.com/rust-lang/rust/issues/59301, which declares what we have to be the lower bound. But I agree it's confusing with respect to the existing functionality. Will rename to `find_lowest_match`.